### DISCUSSION TITLE
Unofficial help for NVIDIA

### DISCUSSION THREAD

**Discussion by sebastichoupinet** [hyprwm/Hyprland#192] [General]
Hi @vaxerski, @viperML , @everyone , 

From what you mentionned in the wiki, you will not support the NVDIA case.
From my point of view, that's fine, I perfectly understand why .
As I've got both NVIDIA and AMD cards, and installed hyprland on both (thanks again), I'm trying to adapt from one case to the other, and I'm sure I won't be the only one.

So, maybe we can open, on this discussion (or on a category of), some threads of NVDIA users.
I've got a few tricks that work, so if it's OK for you, I can share them.
Without any support from you, but it can be useful for some people.

If not, (and again I understand, but had no choice with the second graphic card I own), it's ok for me.
I don't know if this of things can be asked , be nice with me ;)

Thanks again for your amazing work :)

**Comment by fufexan** [comment]
Please, do share what you have. Maybe it'll help users in similar situations.

What I know is that NVIDIA PRIME Offload works, a.k.a the iGPU will be used by Hyprland, and the other card is free for anything else. This is the usual route I recommend to any NVIDIA wayland users, as it's the simplest.

If you know of other methods, feel free to list them here.

**Comment by vaxerski** [comment]
feel free to discuss about nvidia stuff. I won't help, as I don't have an nvidia card, but I agree it'd be help for many.

**Comment by sebastichoupinet** [comment]
OK nice :)

1 -  As an example, some other applications needs x11 or other environment 
As an example , [fusuma](https://github.com/iberianpig/fusuma).
So launching it with some environment variables could help
 -XDG_CURRENT_DESKTOP and XDG_SESSion can lure them, thanks to the gnome-extension
> XDG_CURRENT_DESKTOP=gnome XDG_SESSION_TYPE=x11 fusuma --config=~/.config/fusuma/config-hyprland.yml

2- Invisible cursor = > launch it with  WLR_NO_HARDWARE_CURSORS=1 Hyprland

3- To make it appears on the GDM/whatever Display Manager you use, I didn't find anything

4- Flickering : same

5- copy/paste with middle click; same and very annoying

**Reply by fufexan** [reply]
5 is a general problem, not only on Nvidia.

3 can be achieved by manually writing a session file.

**Reply by Vlad55432** [reply]
Is fourth related to electron/chromium apps? If so, I've seen a lot of flickering with these apps only.
For me, in RTX 3060, it happens, if GPU is idling.
Disabling GPU composition with this flag helps `--disable-gpu-compositing`.
GPU sometimes can still be used, like playing video or something else.

**Reply by MahouShoujoMivutilde** [reply]
@Vlad55432 see https://wiki.hyprland.org/Nvidia/#flickering-in-electron--cef-apps about syncobj, in particularly `--enable-features=WaylandLinuxDrmSyncobj` flag.

It's also `chrome://flags/#wayland-linux-drm-syncobj` in chromium. Needs linux 6.11+ kernel.

No need to disable gpu composing completely.

**Reply by Vlad55432** [reply]
@MahouShoujoMivutilde holy, that worked!
I quite remember, that I was testing enabling syncobj before, but didn't work. Probably electron35/chromium 134 didn't came in time.
I was testing vesktop 1.5.5 and spotify 1:1.2.59.514-1 (maybe?) back then. Now both clients with 1.5.6 and 1:1.2.60.563-2 versions works flawlessly.

**Reply by Vlad55432** [reply]
By the way, I was wrong about Spotify, developers didn't release yet their app with cef 134 or higher version

**Comment by viperML** [comment]
I don't know why you pinged me specifically but still ;)

1,2 and 3 could be fixed in the distribution packages by providing some set of recommended Env variables to be set by Pam. For the nix flake, it would be very easy to do.

As for 4, this seems to be a problem with all Wlroots based compositors. I guess we will have to wait for it to be fixed upstream
